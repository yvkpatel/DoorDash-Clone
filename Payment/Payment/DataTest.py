#Requests relies on having database files downloaded and a connection to database on the local machine
#Therefore this file will not run without an active connection to the database on the local machine it is being run on,
#and as such cannot be tested for MVP
# If you want to run this code for evaluation, the database module is required.


from flask import Flask, request
import requests
import json

def getCustomerFromID(customerID):
    mainURL = "http://localhost:5000/customer/id/"

    r = requests.get(url=(mainURL + str(customerID)))
    cust = r.json()
    return cust



def getRestaurantFromID(restaurantID):
    mainURL1 = "http://localhost:5000/restaurant/id/"

    r1 = requests.get(url=(mainURL1 + str(restaurantID)))
    restaurant= r1.json()
    return restaurant




def getDriverFromID(driverID):
    mainURL2 = "http://localhost:5000/driver/id/"

    r2 = requests.get(url=(mainURL2 + str(driverID)))
    driver = r2.json()
    return driver




#Fetches all unpaid orders from the database
def getOrderFromID():
    mainURL3 = "http://localhost:5000/order/payment/getOrdersUnpaid"

    r3 = requests.get(url=(mainURL3)) # + str(orderID)))
    print(r3)
    if r3.ok:
        return r3.json()
    else:
        return None

#Updates the payment flag within the database to either true or false
def updatePaymentFlag(orderID,isPaid: bool):
    url = "http://localhost:5000/order/payment/update"
    data = {'orderID': orderID, 'paid': isPaid}
    r = requests.post(url, data)
    print(r.json()['reply'])
    return r.json()['reply']




#For testing, fetches all orders from database.
def getAll():
    url = "http://localhost:5000/order/all"
    r = requests.get(url)
    return r.json()

#testing
    #print(getAll())
    #O321764897
    #O252073901
    #O724516036
    #updatePaymentFlag('O724516036','')
    #print(getOrderFromID())



