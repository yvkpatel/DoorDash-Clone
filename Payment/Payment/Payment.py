
#Payment object is created after we are passed info from function call and the string is decoded.
class Payment:

    def __init__(self, user, ccnum, cvc, ccexpiry, postalcode, amtuserpaid):
        self.ccnum = ccnum
        self.user = user
        self.cvc = cvc
        self.ccexpiry = ccexpiry
        self.postalcode = postalcode
        self.amtuserpaid = amtuserpaid

    #Some empty methods, to be implemented if we need to in the future.
    def get_user(self):
        return self.user

    def get_restaurant(self):
        pass

    def get_amt_user_paid(self):
        return self.amtuserpaid

    def get_date_time_of_order(self):
        pass

    def take_our_cut(self):
        #current cut is 20% of order price. TODO: For not MVP, make sure this is done in Algo and is displayed to user beforehand as part of cost.
        self.totalordercost = self.amtuserpaid * 1.2


