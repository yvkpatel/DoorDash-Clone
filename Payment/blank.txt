blank file to create payment folder
