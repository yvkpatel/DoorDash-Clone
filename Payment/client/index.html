<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Stripe Checkout</title>
  </head>
  <body>
    <h1>Checkout with stripe</h1>

    <button type="submit" id="btn">Click me!</button>

    <script src="https://js.stripe.com/v3/"></script>
    <script charset="utf-8">
      var stripe;
      fetch("/public-keys")
        .then((response) => response.json())
        .then((data) => {
          stripe = Stripe(data.key);
        })
        .catch((error) => console.error(error));

      var btn = document.getElementById("btn");
      btn.addEventListener("click", function (e) {
        e.preventDefault();

        fetch("/my-route", {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify({
            test: 123,
            payment_method: "pm_card_visa",
          }),
        })
          .then((response) => response.json())
          .then((data) => {
            console.log("Success:", data);
          })
          .catch((error) => console.error(error));
      });
    </script>
  </body>
</html>
