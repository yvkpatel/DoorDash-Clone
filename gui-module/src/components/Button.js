const Button = ({text, color, onClick}) => {
    return(
        <button className='button'
        style={{background:color}}
        data-testid="button"
        onClick={onClick}>
            {text}
        </button>
    )
}

export default Button