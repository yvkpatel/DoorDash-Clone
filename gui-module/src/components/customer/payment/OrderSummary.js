/**
 * Component to display order summary at checkout
 */
import React from "react";
import OrderList from "./OrderList";
import OrderCard from "./OrderCard";

const OrderSummary = (props) => {
  return (
    <div>
      <h1>Order Summary</h1>
      <OrderList itemList={props.itemList} component={OrderCard} />
      <h2>Total: ${props.cost}</h2>
    </div>
  );
};

export default OrderSummary;
