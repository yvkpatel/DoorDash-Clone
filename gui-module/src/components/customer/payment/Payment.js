/* This component controls the payment information form
 *  Input: function from payment page to handle the submission of payment data
 */

import React from "react";
import { useState } from "react";
import Form from "react-bootstrap/Form";
import Container from "react-bootstrap/Container";
import { Grid, Row, Col } from "react-bootstrap";
import Button from "react-bootstrap/Button";
import { FloatingLabel } from "react-bootstrap";

const Payment = ({ onPay }) => {
  // var visaRegEx = /^(?:4[0-9]{12}(?:[0-9]{3})?)$/;
  // var expiryRegEx = /^(0[1-9]|1[0-2])\/?([0-9]{4}|[0-9]{2})$/;
  // var cVVRegEx = /^[0-9]{3,4}$/;
  //Input fields for payment form
  const [cardNumber, setCardNumber] = useState("");
  const [cardExpiry, setExpiry] = useState("");
  const [cardCVV, setCVV] = useState("");

  //handles the submission of data
  const onSubmit = (e) => {
    e.preventDefault();
    //If details are missing the user is alerted
    if (!cardNumber || !cardExpiry || !cardCVV) {
      alert(
        "Yeah probably best not to put your credit card details into this app..."
      );
      return;
    }

    /*
         if (!visaRegEx.test(cardNumber)) {
             alert('Credit Card number not valid.');
            return;
         }
         if(!expiryRegEx.test(cardExpiry)){
             alert('Credit Card expiry not valid.');
             return;
         }
         if(!cVVRegEx.test(cardCVV)){
             alert('Credit Card CVV not valid.');
             return;
        */
    onPay({ cardNumber, cardExpiry, cardCVV });
    //alert('Lolz we got ur details')
  };
  return (
    <Container className="container2">
      <h1> Payment Information </h1>
      <Form onSubmit={onSubmit}>
        <Row className="mb-3">
          <Form.Group as={Col} controlId="form.Number">
            <Col>
              <FloatingLabel controlId="floatingNum" label="Card Number">
                <Form.Control
                  type="text"
                  value={cardNumber}
                  onChange={(e) => setCardNumber(e.target.value)}
                />
              </FloatingLabel>
            </Col>
          </Form.Group>
        </Row>

        <Row>
          <Form.Group as={Col} className="mb-3" controlId="form.Expiry">
            <FloatingLabel controlId="floatingExpiry" label="Expiry (MM/YY)">
              <Form.Control
                type="text"
                value={cardExpiry}
                onChange={(e) => setExpiry(e.target.value)}
              />
            </FloatingLabel>
          </Form.Group>
          <Form.Group className="mb-3" as={Col} controlId="form.CVV">
            <FloatingLabel controlId="floatingCVV" label="CVV">
              <Form.Control
                type="text"
                value={cardCVV}
                onChange={(e) => setCVV(e.target.value)}
              />
            </FloatingLabel>
          </Form.Group>
        </Row>
        <Button variant="primary" type="submit">
          Submit
        </Button>
      </Form>
    </Container>
  );
};
export default Payment;
