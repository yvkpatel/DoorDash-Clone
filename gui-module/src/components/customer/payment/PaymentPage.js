/**
 * This is the payment page where a customer can see the finished order and submit payment details
 */
import { React, useState, useEffect } from "react";
import Header from "../../Header";
import { Row, Col } from "react-bootstrap";
import Payment from "./Payment";
import OrderSummary from "./OrderSummary";
import Container from "react-bootstrap/Container";
// import PaymentForm from "../components/PaymentForm";
import { useHistory } from "react-router-dom";

const PaymentPage = (props) => {
  //Will pull Order Item List from Database
  const [paydetails, setPayDetails] = useState({});
  const [customerDetails, setCustomer] = useState({});
  const st = props.location.state;
  console.log(props);
  const history = useHistory();

  //Gets order items on page and gets customer details => FAKE SERVER FOR NOW :)
  useEffect(() => {
    // const getItems = async () => {
    //   const itemsFromServer = await fetchItems()
    //   setItems(itemsFromServer)
    // }
    const getCustomer = async () => {
      const custFromServer = await fetchCustomer("C235771756");
    };
    // getItems()
    getCustomer();
  }, []);

  //Deletes object from database after payment. Temporary to reduce frontend database
  // useEffect(()=>{
  //   deletePayment();
  // },[paydetails])

  //Fetch Order Items
  // const fetchItems = async () => {
  //   const res = await fetch('http://localhost:5000/items')
  //   const data = await res.json()

  //   return data
  // }

  //Fetch Customer Details
  const fetchCustomer = (id) => {
    const res = fetch("/customer/id/" + id)
      .then((resp) => resp.json())
      .then((data) => setCustomer(data))
      .catch((err) => console.log(err));
  };

  //This creates a JSON file as requested by the payment team
  const makePayJson = (props) => {
    var obj = new Object();
    //TODO change obj items to all required things
    obj.ccv = props.cardCVV;
    obj.card_number = props.cardNumber;
    obj.expiry = props.cardExpiry;

    obj.name = customerDetails.name;
    obj.postal_code = customerDetails.postal_code;
    //TODO: fill with real info from backend
    obj.total = "$420.69";
    var JSONstring = JSON.stringify(obj, null, 2);

    //print (just for testing) - REMOVE LATER
    alert(JSONstring);
    return JSONstring;
  };

  // const sendPayment = async (props) => {
  //   const payJson = makePayJson(props);
  //   // const res = await fetch('http://localhost:5000/payment', {
  //   //   method: 'POST',
  //   //   headers: {
  //   //     'Content-type': 'application/json'
  //   //   },
  //   //   body: payJson,
  //   // })
  //   // const data = await res.json();
  //   //setPayDetails(data);
  //   console.log(paydetails);
  //   return data;
  // }

  // const deletePayment = async () => {
  //   await fetch('http://localhost:5000/payment/1', {
  //     method: 'DELETE',
  //   })
  // }

  //TODO: pull from the backend
  var cost = 0;
  var items = [];
  if (st != null) {
    cost = st.cart.price;
    items = st.cart.itemList;
  }

  //This function will handle how payment of the order is handled
  const submitPayment = (cardDetails) => {
    console.log(customerDetails._Customer__name);
    history.push("../customermap");
    // const data = sendPayment(cardDetails);
  };

  return (
    <div className="App" data-testid="payment-page">
      <Header />
      <Container>
        <Row>
          <Col xs={12} md={8}>
            {" "}
            {/*  Component to display order summary */}
            <OrderSummary itemList={items} cost={cost} />
          </Col>
          <Col xs={6} md={4}>
            {" "}
            {/*  Component to display payment form */}
            {/* <PaymentForm/> */}
            <Payment onPay={submitPayment} />
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default PaymentPage;
