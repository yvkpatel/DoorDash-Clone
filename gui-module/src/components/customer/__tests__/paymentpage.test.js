import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import React, { useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import ReactDOM from "react-dom";
import Button from "../../Button";

import PaymentPage from "../payment/PaymentPage";
import OrderSummary from "../payment/OrderSummary";
import Payment from "../payment/Payment";
import "@testing-library/jest-dom/extend-expect";

afterEach(() => {
  cleanup();
});

const MockOrderSummary = {
  location: {
    state: {
      cart: {
        itemList: [
          {
            id: 1,
            title: "Fries",
            price: 3.55,
          },
        ],
      },
    },
  },
};

test("should render customer payment component", () => {
  render(
    <BrowserRouter>
      <PaymentPage {...MockOrderSummary} />
    </BrowserRouter>
  );
  const PaymentPageElement = screen.getByTestId("payment-page");
  expect(PaymentPageElement).toBeDefined();
});

test("should should show order summary", () => {
  const { getByText } = render(
    <OrderSummary
      itemList={MockOrderSummary.location.state.cart.itemList}
      price="3.55"
    />
  );

  expect(getByText("Fries")).toBeTruthy();
});

test("should should show payment information field", () => {
  const { getByText } = render(<Payment />);

  expect(getByText("Payment Information")).toBeTruthy();
});

test("matches snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <PaymentPage {...MockOrderSummary} />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
