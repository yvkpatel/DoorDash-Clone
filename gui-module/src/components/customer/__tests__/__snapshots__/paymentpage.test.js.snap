// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="payment-page"
>
  <header
    className="header"
  >
    <img
      alt="YUMI Logo"
      className="logo"
      src="logo-header.png"
    />
    <a
      className="accSettings"
      href="https://developer.mozilla.org/en-US/docs/Web/CSS"
    >
      Account Settings
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "background": "#2DB6F0",
          }
        }
      >
        Logout
      </button>
    </a>
  </header>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-8 col-12"
      >
         
        <div>
          <h1>
            Order Summary
          </h1>
          <div
            className="list"
          >
            <div
              className="orderCustomer"
            >
              <div
                className="card"
              >
                <div
                  className="card-body"
                >
                  <div
                    className="card-title h5"
                  >
                    Fries
                  </div>
                  <p
                    className="card-text"
                  >
                    $
                    3.55
                  </p>
                </div>
              </div>
            </div>
          </div>
          <h2>
            Total: $
          </h2>
        </div>
      </div>
      <div
        className="col-md-4 col-6"
      >
         
        <div
          className="container2 container"
        >
          <h1>
             Payment Information 
          </h1>
          <form
            className=""
            onSubmit={[Function]}
          >
            <div
              className="mb-3 row"
            >
              <div
                className="col"
              >
                <div
                  className="col"
                >
                  <div
                    className="form-floating"
                  >
                    <input
                      className="form-control"
                      id="floatingNum"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="floatingNum"
                    >
                      Card Number
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="mb-3 col"
              >
                <div
                  className="form-floating"
                >
                  <input
                    className="form-control"
                    id="floatingExpiry"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    htmlFor="floatingExpiry"
                  >
                    Expiry (MM/YY)
                  </label>
                </div>
              </div>
              <div
                className="mb-3 col"
              >
                <div
                  className="form-floating"
                >
                  <input
                    className="form-control"
                    id="floatingCVV"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    htmlFor="floatingCVV"
                  >
                    CVV
                  </label>
                </div>
              </div>
            </div>
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
